### DISCUSSION TITLE
Que tipo de front-end developer você é?

### DISCUSSION THREAD

**Discussion by dionarodrigues** [frontendbr/forum#1824] [Discussão]
Se você, assim como eu, veio da área de design (e costumava usar o termo web designer para se referir a si mesmo como profissional), talvez possa se questionar qual tipo de desenvolvedor(a) você é, uma vez que cada vez mais o desenvolvimento front-end lida com códigos complexos e bem próximos do back-end. 

E essa confusão é "normal" já que um único desenvolverdor front-end pode ter habilidades tanto de design quanto de back-end, pois ele está no meio desses dois extremos. Brad Frost, criador do Atomic Design, também passou por essa crise e, inclusive, escreveu esse [post sobre esse assunto](https://bradfrost.com/blog/post/frontend-design/).

Existem hoje muitos rótulos para o profissional da nossa área, no entanto, vejo que dois deles se sobressaem e geram algumas discussões: front-end designer e front-end engineer.  Inclusive, Chris Coyier (criador do CSS-Tricks), levantou a discussão sobre isso nesse [outro post](https://css-tricks.com/the-great-divide/). 

**Enquanto o Front-end Designer se preocupa mais com HTML, CSS, design, interação, padrões, acessibilidade etc, o Front-end Engineer está fortemente relacionado com o JavaScript.** 

**Então eu te pergunto, qual tipo de desenvolvedor(a) front-end você é? Front-end Designer ou Front-end Engineer?** 🤷‍♀️😎

Escrevi um post sobre isso onde eu estendo um pouco mais as explicações e coloco o meu ponto de vista. [Confira o meu artigo "Que tipo de desenvolvedor front-end eu sou" aqui](https://www.diogorodrigues.dev/pt/blog/que-tipo-de-desenvolvedor-frontend-eu-sou).

**Comment by theandersonn** [comment]
@diogorodrigues Parabéns pelo novo site, ficou elegante e funcional. 

(_Já citei seu trabalho para várias pessoas. Entre as habilidades que eu considero difícil unir é Design e Desenvolvimento. E você faz isso com excelência, parabéns!_)

Eu venho da época dos Web Designers, cara... como mudou!

Hoje o trabalho que eu faço e o direcionamento que tenho tomado na minha carreira está mais para engenheiro, de acordo com os posts e as ideias em torno desse ecossistema. Mas sinceramente eu me sinto desconfortável com esse termo, não sei se é a nossa cultura, nossa forma de enxergar as tarefas atribuídas para um engenheiro.

Particularmente para mim, entre as responsabilidades de um Engenheiro de Software estão: definir e monitorar questões relacionadas à Agilidade, Arquitetura, Padrões e Testes. 

Sim, desenvolvedores também tem essas responsabilidades, mas vejo o Engenheiro num escopo de supervisão.

É uma opinião minha, sinceramente estou cunhando, a medida que o próprio mercado está me ensinando.

**Comment by tarciozemel** [comment]
Tinha visto esse artigo no CSS-Tricks; tava pensando em traduzir. Seu artigo ficou muito legal, muito bem formatado.

Dentro da proposta da coisa, com certeza sou "Front-end Designer". MAS...

Na verdade, na verdade, acho essa divisão meio embriaguez de sucesso. Pessoal toma nomes de profissões consagradas (Arquiteto, Engenheiro etc.) para afagar o ego; gosta de dividir e dividir cada vez mais as áreas de atuação para falar que é "especializado".

Qual seria o problema com "Desenvolvedor front-end focado em JavaScript"?

**Comment by dionarodrigues** [comment]
@theandersonn Muito obrigado! É muito gratificante receber elogios e saber que você gosta do meu trabalho. ✌

Pois é, a nossa área vem mudando bastante e creio que continuará passando por diversas transformações, uma vez que cada vez mais surgem nossas maneiras de fazer desenvolvimento. O que é bom, claro. Mas por outro lado também faz com que aumente a possibilidade dessa crise de identidade como profissional aparecer em algum momento em nossas vidas. Você chegou a passar por isso? Ou se viu como front-end engineer de modo tranquilo?

@tarciozemel Fico feliz com seu feedback, muito obrigado! 😁 
Tb me considero um front-end designer.

Entendo seu ponto de vista, mas acredito que esses rótulos surgem da necessidade de encontrar nomes adequados às transformações que o desenvolvimento vem passando, assim como também vem acontecendo na área do design e em tantas outras que permitem esse tipo de flexibilização. Ao contrário disso, talvez ainda teríamos apenas o termo "programador" como se tinha há muitos anos.

Acho que o termo "programador" e "web designer", por exemplo, tem ficado cada vez mais distantes dos anúncios de vagas atuais. Enquanto que "front-end developer", na maior parte das vezes está fortemente ligado com o ecossistema javascript (muitas vezes apenas com back-end) e ainda superficialmente com o design. Por isso, eu vejo esses incentivos de nomes como algo positivo quando olhados pela perspectiva do mercado de trabalho. E, claro, seria ótimo um consenso/padrão sobre esses rótulos num futuro.

**Comment by evertonthepaula** [comment]
Programador nasce programador, designer nasce designer. É possível ensinar padrões de desenvolvimento para um designer e padrões gráficos para um programador, mas a criatividade lógica para criar códigos e a criatividade artística para criar interfaces não pode ser ensinada, ou você tem essa habilidade e consegue evoluir ou não tem. No caso eu não tenho muita habilidade artística, apesar de trabalhar com design de interação e UX a parte de UI não é comigo, não mesmo. Hoje eu me considero um "front-end developer" igual todo mundo, mas gosto de automarizar as coisas, isto me faz um "Front-End Dev. Ops" ?  Vejo o engenheiro front-end como sendo a pessoa que  implementa engenharia de software ao front-end, você tem que saber escalar CSS(principalmente sem chorar e reclamar), HTML, Componentes, Javascript e Arquitetura como ninguém. Pelo domínio em engenharia de software, o "front-end engineer" talvez seja a melhor pessoa para um líder técnico.

Mas o melhor lugar para responder este tipo de pergunta ainda é o front-end-handbook:  https://frontendmasters.com/books/front-end-handbook/2019/#3.2

**Comment by ninetails** [comment]
Taxonomias foram criadas pra classificar "coisas".

Mas quando levada ao pessoal, minha opinião é que cada um se denomina como quer (e no que se garante).

Sempre curti dizer que sou **desenvolvedor web** e se alguém pedir alguma explicação a mais, digo **frontend**. Não quer dizer que não sei outras coisas, nem também que sou Unicórnio ( #506 ), longe disso. Tem coisa que sei, tem coisa que não sei; tem coisa que tenho afinidades e gostos, também tem coisa que eu deixo de lado no crescimento profissional, guiados pelo meu interesse pessoal. Posso gostar de programar em JS; não gostar mas saber sobre backend a ponto de preferir nem citar ou aprofundar estudos; me preocupar com standards... tem coisa que eu sei e gosto e tem muito mais coisa que não sei e/ou não gosto.

Só acho que denominações "binárias" são como conversas de boteco sobre política: todo mundo curte dar uma polarizada à situação e inclusive não acho nem que carreiras seguem um espectro... Explicando por um viés que qualquer um que trabalhou com ~~Wordpress~~ CMS vai se sentir familiarizado: é mais pertinente a gente pensar nisso como uma taxonomia por _tags_ que por categorias. Mesmo pq só falando sobre frontend, é uma tecnologia nova que surge todo mês :laughing: , imagino que outras áreas de TI o mesmo. Não dá pra nascer sabendo tudo.

Acho válido essa categorização só na hora de oportunidades de emprego, pois aí vc especifica o perfil procurado e precisa ser necessariamente objetivo nestes anúncios.

**Comment by tarciozemel** [comment]
> Entendo seu ponto de vista, mas acredito que esses rótulos surgem da necessidade de encontrar nomes adequados às transformações que o desenvolvimento vem passando, assim como também vem acontecendo na área do design e em tantas outras que permitem esse tipo de flexibilização. Ao contrário disso, talvez ainda teríamos apenas o termo "programador" como se tinha há muitos anos.

Se você pegar o título "Front-end Designer", não dá a entender que mexe com Design de Interação, SVG, CSS/Sass, Arquitetura CSS, temas WordPress, jQuery, UX, a11y etc., como sugere a divisão do artigo...  ¯\_(ツ)_/¯

Entendi o que você quis dizer, mas, se realmente for isso em detrimento à embriaguez de sucesso (o que eu acho que _não é_), então ainda têm muuuito trabalho pela frente para chegar a um nome adequado.

**Comment by dionarodrigues** [comment]
Interessante esse site que você compartilhou @evertonthepaula . Inclusive, além dos Job Titles, tem um monte de informação importante. Valeu!

Entendo o seu ponto de vista e concordo que há sim pessoas com mais facilidade em uma área que em outra. Mas não acredito que "ou se é designer ou se é programador" seja uma verdade absoluta, uma vez que há muitos estudos comprovando que a criatividade pode ser estimulada e que qualquer habilidade pode ser adquirida através de prática e disciplina. O Filipe Deschamps, inclusive, discute sobre pessoas que acreditam não possuírem o dom de programar [nesse vídeo](https://www.youtube.com/watch?v=ZtMzB5CoekE), e coloca um ponto de vista interessante em debate.

No meu ponto de vista, penso que é mais uma questão de identificação, pois quando nos identificamos com algo, acabamos nos esforçando mais naquele sentido. Por exemplo, eu sempre me identifiquei mais com design, assim, acabo gastando muito mais tempo com assuntos relacionados à design que programação. E você, pelo que me diz, se identifica mais com programação.

---

Concordo @ninetails, acho interessante essa questão que cada um escolhe a label que mais se identifica, até porque o mercado de trabalho funciona assim também.

---

@tarciozemel Eu vejo "Front-end Designer" não como uma descrição minuciosa das habilidades do profissional, mas como algo mais abrangente entre os dois extremos que o front-end se encontra (back-end e design).

**Comment by tarciozemel** [comment]
@diogorodrigues O ponto não é ser uma "descrição minuciosa", mas se vamos começar a dar nomes a fim de especificar/especializar, eles têm que, minimamente, fazer alusão a seus objetos/objetivos principais.

Do contrário, posso sugerir que "Cintesbrósni" significa o profissional de Web que mexe com Design de Interação, SVG, CSS/Sass, Arquitetura CSS, temas WordPress, jQuery, UX, a11y...

Convença-me do contrário. Até o momento, parece que a coisa toda é #embriaguezdesucesso.

**Comment by dionarodrigues** [comment]
Tranquilo @tarciozemel . Entendo o seu ponto de vista.
